modules/frontend/frontend.go

```diff
@@ -239,20 +245,19 @@ func newMetricsSummaryHandler(next pipeline.AsyncRoundTripper[combiner.PipelineR
 // prepareRequestForQueriers modifies the request so they will be farmed correctly to the queriers
 //   - adds the tenant header
 //   - sets the requesturi (see below for details)
-func prepareRequestForQueriers(req *http.Request, tenant string, originalURI string, params url.Values) {
+func prepareRequestForQueriers(req *http.Request, tenant string) {
```

Comment:
I find this function pretty obscure. Why don't we return a new http.Request based on the original one?

```go
func getQuerierHTTPRequest(origin_req *http.Request, tenant string) *http.Request{
}
```

That way we will reduce the cognitive overload reading the code

Reply:
I would prefer modifying in place to reduce allocs